pkg/reconciler/pubsub/resources/iam_policy_bindings.go

```diff
@@ -0,0 +1,100 @@
+/*
+Copyright 2020 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resources
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/google/knative-gcp/pkg/utils"
+	"google.golang.org/api/iam/v1"
+	corev1 "k8s.io/api/core/v1"
+)
+
+// AddIamPolicyBinding will add iam policy binding, which is related to a provided k8s ServiceAccount, to a GCP ServiceAccount.
+func AddIamPolicyBinding(ctx context.Context, projectID string, gServiceAccount *string, kServiceAccount *corev1.ServiceAccount) error {
+	if err := SetIamPolicy(ctx, "add", projectID, gServiceAccount, kServiceAccount); err != nil {
+		return fmt.Errorf("failed to add iam policy binding: %w", err)
+	}
+	return nil
+}
+
+// RemoveIamPolicyBinding will remove iam policy binding, which is related to a provided k8s ServiceAccount, from a GCP ServiceAccount.
+func RemoveIamPolicyBinding(ctx context.Context, projectID string, gServiceAccount *string, kServiceAccount *corev1.ServiceAccount) error {
+	if err := SetIamPolicy(ctx, "remove", projectID, gServiceAccount, kServiceAccount); err != nil {
+		return fmt.Errorf("failed to remove iam policy binding: %w", err)
+	}
+	return nil
+}
+
+func SetIamPolicy(ctx context.Context, action, projectID string, gServiceAccount *string, kServiceAccount *corev1.ServiceAccount) error {
+	iamService, err := iam.NewService(ctx)
```

Comment:
this will be hard to UT because we are not mocking it... 
There are some examples of how we do mocking of these libraries, although it might require some effort in this case. 
Can you create an issue and add a TODO here saying that this should be mocked and link to the issue?

Reply:
> There are some examples of how we do mocking of these libraries, although it might require some effort in this case.

Are you referencing the testing client in pkg/gclient?